lib/svg_sprite/svg_sprite.rb

```diff
@@ -415,28 +421,34 @@ def self.prepare_symbol(symbol, svg_filename)
   end
 
   def self.settings_icons
-    # includes svg_icon_subset and any settings containing _icon (incl. plugin settings)
-    site_setting_icons = []
+    get_set_cache("settings_icons") do
```

Comment:
Expiry of cache is done in https://github.com/discourse/discourse/blob/eb2c399445bf1007c9eb3c358003191ffcec0211/config/initializers/014-track-setting-changes.rb#L42